htdocs/core/login/functions_ldap.php

```diff
@@ -245,18 +246,20 @@ function check_user_password_ldap($usertotest, $passwordtotest, $entitytotest)
 			 ** 53 - Account inactive (manually locked out by administrator)
 			 */
 			dol_syslog("functions_ldap::check_user_password_ldap Authentication KO failed to connect to LDAP for '".$usertotest."'", LOG_NOTICE);
-			if (is_resource($ldap->connection) || is_object($ldap->connection)) {    // If connection ok but bind ko
-				$ldap->ldapErrorCode = ldap_errno($ldap->connection);
-				$ldap->ldapErrorText = ldap_error($ldap->connection);
-				dol_syslog("functions_ldap::check_user_password_ldap ".$ldap->ldapErrorCode." ".$ldap->ldapErrorText);
+			if ($ldapdebug) {
+				if ($ldap->connection != false) {
+					if (is_resource($ldap->connection) || is_object($ldap->connection)) {    // If connection ok but bind ko
+						$ldap->ldapErrorCode = ldap_errno($ldap->connection);
+						$ldap->ldapErrorText = ldap_error($ldap->connection);
+						dol_syslog("functions_ldap::check_user_password_ldap ".$ldap->ldapErrorCode." ".$ldap->ldapErrorText);
```

Comment:
The dol_syslog here must always be called on connection error, because it is used by some fail2ban profile to detect abuse.
See file dev/setup/fail2ban/filter.d/web-dolibarr-rulesbruteforce.conf